Index: boinc-app-eah-brp/Makefile
===================================================================
--- boinc-app-eah-brp.orig/Makefile
+++ boinc-app-eah-brp/Makefile
@@ -78,14 +78,21 @@ release: CXXFLAGS += -DNDEBUG -DLOGLEVEL
 #   LDFLAGS += $(CXXFLAGS)
 #endif
 
+WUzap=./debian/extra/einstein_bench/testwu/p2030.20151015.G187.41-00.88.N.b2s0g0.00000.zap
+WUbin=./debian/extra/einstein_bench/testwu/p2030.20151015.G187.41-00.88.N.b2s0g0.00000_1099.bin4
+WUbank=./debian/extra/einstein_bench/testwu/stochastic_full.bank
+
+$(WUbank).50:	$(WUbank)
+	head -n 50 $< > $@
 
 # file based targets
-profile:
+profile: results_profile.cand
+results_profile.cand: $(TARGET) $(WUbank).50
 	@echo "Removing previous profiling data..."
 	rm -f *_profile.*
 	rm -f *.gcda
 	@echo "Gathering profiling data (this takes roughly one minute)..."
-	./$(TARGET) -t $(EINSTEIN_RADIO_SRC)/../test/templates_400Hz_2_short.bank -l $(EINSTEIN_RADIO_SRC)/data/zaplist_232.txt -A 0.04 -P 3.0 -W -z -i $(EINSTEIN_RADIO_SRC)/../test/J1907+0740_dm_482.binary -c status_profile.cpt -o results_profile.cand
+	./$(TARGET) -t $(WUbank).50 -l $(WUzap) -A 0.04 -P 3.0 -W -z -i $(WUbin) -c status_profile.cpt -o results_profile.cand
 	@echo "Finished gathering profiling data..."
 
 $(TARGET): $(DEPS) $(EINSTEIN_RADIO_SRC)/erp_boinc_wrapper.cpp $(OBJS)
