Index: boinc-app-eah-brp/demod_binary_resamp_cpu.c
===================================================================
--- boinc-app-eah-brp.orig/demod_binary_resamp_cpu.c
+++ boinc-app-eah-brp/demod_binary_resamp_cpu.c
@@ -39,11 +39,11 @@ float *del_t = NULL;
 int set_up_resampling(DIfloatPtr input_dip, DIfloatPtr *output_dip, const RESAMP_PARAMS *const params, float *sinLUTsamples, 
 float *cosLUTsamples)
 {
-    float * input = input_dip.host_ptr;
+    //float * input = input_dip.host_ptr;
     float ** output = & (output_dip->host_ptr);
 
     // unused
-    input = NULL;
+    //input = NULL;
     sinLUTsamples = NULL;
     cosLUTsamples = NULL;
 
