Index: boinc-app-eah-brp/erp_boinc_ipc.cpp
===================================================================
--- boinc-app-eah-brp.orig/erp_boinc_ipc.cpp
+++ boinc-app-eah-brp/erp_boinc_ipc.cpp
@@ -34,13 +34,13 @@ using namespace std;
 #include "boinc_api.h"
 #include "util.h"
 
-BOINC_OPTIONS erp_boinc_options;
-APP_INIT_DATA erp_app_init_data;
+static BOINC_OPTIONS erp_boinc_options;
+static APP_INIT_DATA erp_app_init_data;
 t_pulsar_search erp_search_info;
 
 #include "graphics2.h"
 
-char* shmem = NULL;
+static char* shmem = NULL;
 
 void erp_update_shmem(void)
 {
