Index: boinc-app-eah-brp/demod_binary.c
===================================================================
--- boinc-app-eah-brp.orig/demod_binary.c
+++ boinc-app-eah-brp/demod_binary.c
@@ -1465,7 +1465,10 @@ thrA[4] = 0.0;
 
 #ifdef BOINCIFIED
       // update work unit fraction done (counter is 0-indexed)
-      erp_fraction_done((template_counter + 1.0) / template_total_amount);
+#ifdef COMMUNICATIONREDUCTION
+      if (0 == template_counter % COMMUNICATIONREDUCTION)
+#endif
+         erp_fraction_done((template_counter + 1.0) / template_total_amount);
 #endif
 
 #ifndef NOCHECKPOINTING
Index: boinc-app-eah-brp/erp_boinc_wrapper.cpp
===================================================================
--- boinc-app-eah-brp.orig/erp_boinc_wrapper.cpp
+++ boinc-app-eah-brp/erp_boinc_wrapper.cpp
@@ -199,7 +199,7 @@ gzFile boinc_gzopen(const char* path, co
   return gzdopen(fileno(fp), mode);
 }
 
-void erp_fraction_done(double frac) {
+void erp_fraction_done(const double frac) {
   boinc_fraction_done((frac+pass)/passes);
 }
 
Index: boinc-app-eah-brp/erp_boinc_ipc.h
===================================================================
--- boinc-app-eah-brp.orig/erp_boinc_ipc.h
+++ boinc-app-eah-brp/erp_boinc_ipc.h
@@ -53,7 +53,7 @@ extern "C" {
 
     extern void erp_boinc_init(void);
 
-    extern void erp_fraction_done(double);
+    extern void erp_fraction_done(const double);
 #ifdef __cplusplus
 }
 #endif
