Index: boinc-app-eah-brp/demod_binary.c
===================================================================
--- boinc-app-eah-brp.orig/demod_binary.c
+++ boinc-app-eah-brp/demod_binary.c
@@ -1109,8 +1109,12 @@ int MAIN (unsigned int argc, char *argv[
       return(RADPUL_EVAL);
     }
 
+  /*****
+   * commented out: the result of this function is not used
+   *****/
+
   // initialize sin/cos lookup tables (required during resampling)
-  sincosLUTInitialize(&sinLUTsamples, &cosLUTsamples);
+  //sincosLUTInitialize(&sinLUTsamples, &cosLUTsamples);
 
   // prepare resampling
   RESAMP_PARAMS params;
